core/state_processor.go

```diff
@@ -739,6 +715,74 @@ func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (ty
 	return receipts, emittedEtxs, allLogs, statedb, *usedGas, *usedState, multiSet, nil
 }
 
+// RedeemLockedQuai redeems any locked Quai for coinbase addresses at specific block depths.
+// It processes blocks based on predefined lockup periods and checks for unlockable Quai.
+// This function is intended to be run as part of the block processing.
+func RedeemLockedQuai(hc *HeaderChain, statedb *state.StateDB, currentBlockHeight *big.Int) error {
+	// Array of specific block depths for which we will redeem the Quai
+	blockDepths := []uint64{
+		params.ConversionLockPeriod,      // Minimum lockup period
+		params.LockupByteToBlockDepth[0], // 2 hours
+		params.LockupByteToBlockDepth[1], // 4 hours
+		params.LockupByteToBlockDepth[2], // 8 hours
+		params.LockupByteToBlockDepth[3], // 12 hours
+	}
+
+	// Loop through the predefined block depths
+	for _, blockDepth := range blockDepths {
+		// Ensure we can look back far enough
+		if currentBlockHeight.Uint64() <= blockDepth {
+			// Skip this depth if the current block height is less than or equal to the block depth
+			break
+		}
+
+		// Calculate the target block height by subtracting the blockDepth from the current height
+		targetBlockHeight := new(big.Int).Sub(currentBlockHeight, new(big.Int).SetUint64(blockDepth))
+
+		// Fetch the header and block at the calculated target height
+		targetHeader := hc.GetHeaderByNumber(targetBlockHeight.Uint64())
+		if targetHeader == nil {
+			return fmt.Errorf("header at height %d not found", targetBlockHeight.Uint64())
+		}
+
+		// Fetch the block at the calculated target height
+		targetBlock := hc.bc.GetBlock(targetHeader.Hash(), targetBlockHeight.Uint64())
+		if targetBlock == nil {
+			return fmt.Errorf("block at height %d not found", targetBlockHeight.Uint64())
+		}
+
+		for _, etx := range targetBlock.Body().ExternalTransactions() {
```

Comment:
These are outbound ETXs. You cannot execute outbound ETXs until they go through Prime (or Region, respectively)